.vitepress/config.mts

```diff
@@ -11,13 +11,21 @@ const files = globSync('./src/**/README.md').map(item => {
     text: name,
     link: item
   }
-}).filter(Boolean)
+}).filter(notEmpty)
+
+function notEmpty<TValue>(value: TValue | null | undefined): value is TValue {
+  return value !== null && value !== undefined
+}
```

Comment:
Any reason not to use

`.filter(<T,>(val?: T): val is T => Boolean(val))`

here?

It's closer to the JS and to me says "I'm only doing this cos Typescript can't figure this out on its own"

Lastly, could you make sure this file is picked up by `vue-tsc` if we are doing these sorts of things here? ty!

Reply:
> Any reason not to use
>
> `.filter(<T,>(val?: T): val is T => Boolean(val))`
>
> here?

I don’t see how that’s any better.

> Lastly, could you make sure this file is picked up by vue-tsc if we are doing these sorts of things here?

This should definitely not be picked up by our `lint:ts` script because that should only evaluate source code. We could use a separate vue-tsc script, but this feels like over-engineering. Not sure what the value of that would be.